[CI] Add a workflow for quick perf comparison

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96166

Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

desertfire committed:
[CI] Add a workflow for quick

Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Add ciflow/inductor-perf-compare for
PRs that need to do a faster performance sanity check (--training --amp).

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Add a workflow for quick"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Add ciflow/inductor-perf-compare for
PRs that need to do a faster performance sanity check (--training --amp).

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

malfet commented:
@desertfire looks like it's a chicken and egg problem right now: bot only recognizes labels that exists on trunk. I'll try to submit a fix sometime next week, but in the meantime, just push the tag by hand

weiwangmeta commented:
We now have small/medium/large inductor tests with this PR, great! 
To support this medium workflow, we may need to monitor the capacity. Since the nightly will trigger towards the end of the day, it may be beneficial to increase the (linux.gcp.a100.large) runner pool size from 4 to 8, as after around 6pm/7pm, this job would be queueing for a long time.

desertfire committed:
Update on "[CI] Add a workflow for fast inductor perf collection"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Add ciflow/inductor-perf-compare for
PRs that need to do a faster performance sanity check (--training --amp).

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

huydhn commented on .github/workflows/upload-test-stats.yml:
```diff
@@ -2,7 +2,7 @@ name: Upload test stats
 
 on:
   workflow_run:
-    workflows: [pull, trunk, periodic, inductor, inductor-A100-perf]
```
You don't need `inductor-A100-perf-nightly` both in `upload-test-stats.yml` and in `upload-torch-dynamo-perf-stats`.  Once you have the commit in https://github.com/pytorch/pytorch/pull/96165, you will just need to update the latter `upload-torch-dynamo-perf-stats.yml`

desertfire commented:
> We now have small/medium/large inductor tests with this PR, great! To support this medium workflow, we may need to monitor the capacity. Since the nightly will trigger towards the end of the day, it may be beneficial to increase the (linux.gcp.a100.large) runner pool size from 4 to 8, as after around 6pm/7pm, this job would be queueing for a long time.

For now, I have changed my PR to use the linux.gcp.a100 runner pool. Given that smoke tests are small and this perf comparison workflow is only triggered by manually adding the label, we should be ok.

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

weiwangmeta commented:
> 

Sure, I will monitor the queue and increase the max capacity if needed.

weiwangmeta reviewed (approved):
LGTM. How long do we expect the end to end time assuming 4 parallel runners? Is it still about 1 hour each?

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

desertfire commented:
> LGTM. How long do we expect the end to end time assuming 4 parallel runners? Is it still about 1 hour each?

Yes, the run is within 1 hour.

desertfire committed:
Update on "[CI] Add a workflow for quick perf comparison"


Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

[ghstack-poisoned]

desertfire commented:
@pytorchbot rebase

desertfire commented:
@pytorchbot merge -f "added tests have passed and buck-build-test fails on HUD"
